From f1f72da2c7aa9694b401194c3ac2b6fa472b91fa Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 24 Apr 2020 09:28:36 -0700
Subject: [PATCH] CMakeLists.txt: fix build without catkin testing enabled

* use CATKIN_ENABLE_TESTING conditional as suggested in
  http://docs.ros.org/jade/api/catkin/html/howto/format2/gtest_configuration.html
* otherwise do_configure fails with:
  | CMake Error at /jenkins/mjansa/build/ros/webos-melodic-honister/tmp-glibc/work/qemux86-webos-linux/flir-ptu-driver/0.2.1-1-r0/recipe-sysroot/usr/opt/ros/melodic/share/roslint/cmake/roslint-extras.cmake:67 (catkin_run_tests_target):
  |   Unknown CMake command "catkin_run_tests_target".
  | Call Stack (most recent call first):
  |   CMakeLists.txt:44 (roslint_add_test)
  and
  | CMake Error at /jenkins/mjansa/build/ros/webos-melodic-honister/tmp-glibc/work/qemux86-webos-linux/flir-ptu-driver/0.2.1-1-r0/recipe-sysroot/usr/opt/ros/melodic/share/roslaunch/cmake/roslaunch-extras.cmake:66 (catkin_run_tests_target):
  |   Unknown CMake command "catkin_run_tests_target".
  | Call Stack (most recent call first):
  |   CMakeLists.txt:47 (roslaunch_add_file_check)

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 66cb9e3..488ccf2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,6 +41,8 @@ install(DIRECTORY launch
 
 ## Tests
 roslint_cpp()
+if (CATKIN_ENABLE_TESTING)
 roslint_add_test()
 roslaunch_add_file_check(launch/ptu.launch
   DEPENDENCIES flir_ptu_example_urdf flir_ptu_node)
+endif()
